ludwig/models/model.py

```diff
@@ -1110,8 +1113,29 @@ def collect_weights(
         pass
 
     def save_weights(self, save_path):
-        # todo tf2: reintroduce functionality
-        #self.weights_save_path = self.saver.save(save_path)
+        # collect all custom metrics used in output features
+        custom_objects = {}
+        for of_name, of in self.ecd.output_features.items():
+            for mfn_name, mfn_obj in of.metric_functions.items():
+                # if module name starts with 'ludwig' this is a custom metric
+                if mfn_obj.__class__.__module__[:6] == 'ludwig':
+                    custom_objects.update(
```

Comment:
why update and not just:
```
custom_objects[mfn_obj.__class__.__name__] = mfn_obj.__class__
```
?

Reply:
From testing it appears only Ludwig custom metric functions need to be recorded.  The standard Keras metrics are appear to be handled on `load_model`.   I can certainly give it a try and save all metrics and see how it affects `load_model()` operation.

Reply:
Maybe I wasn't clear sorry. What i mean is why using dictionary update function instead of just assigning the value to the dictionary.